### PR TITLE
feat: Support Trino 440

### PR DESCRIPTION
**Pull request by sbernauer**
# Description

Needs https://github.com/stackabletech/docker-images/pull/587

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [ ] CRD changes approved
- [ ] CRD documentation for all fields, following the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs-style-guide).
- [ ] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
- [ ] Changes need to be "offline" compatible
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] Code contains useful logging statements
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated. Follows the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs-style-guide).
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
- [ ] [Roadmap](https://github.com/orgs/stackabletech/projects/25/views/1) has been updated
```

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -22,6 +22,7 @@ All notable changes to this project will be documented in this file.
 [#501]: https://github.com/stackabletech/trino-operator/pull/501
 [#531]: https://github.com/stackabletech/trino-operator/pull/531
 [#539]: https://github.com/stackabletech/trino-operator/pull/539
+[#552]: https://github.com/stackabletech/trino-operator/pull/552
```
Comment (maltesander) [inline CHANGELOG.md:25]: 539 ref can be removed?

**Review by maltesander — DISMISSED** [review]
Minor thing, LGTM otherwise!

File: CHANGELOG.md (hunk 1 above)
Comment (sbernauer) [reply CHANGELOG.md:24]: Good catch!

File: rust/operator-binary/src/config/jvm.rs [hunk 2]
```diff
@@ -29,7 +29,7 @@ pub enum Error {
     },
 }
 
-// Currently works for all supported versions (414, 428, 439 as of 2024-02-08) but maybe be changed
+// Currently works for all supported versions (414, 428, 440 as of 2024-02-08) but maybe be changed
```
Comment (maltesander) [inline rust/operator-binary/src/config/jvm.rs:32]:
```suggestion
// Currently works for all supported versions (414, 428, 440 as of 2024-03-11) but maybe be changed
```

**Comment by sbernauer** [comment]
https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/trino-operator-it-custom/115/

**Review by maltesander — APPROVED** [review]
LGTM if test succeeds!

**Comment by sbernauer** [comment]
https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/trino-operator-it-custom/116/

**Pull request merged**